tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,263 @@
+            CONF_HOST: HOST,
+            CONF_PORT: PORT,
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+        }
+
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_form_timeout(hass):
+    """Test we handle server search timeout."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover",
+        mock_failed_discover,
+    ), patch("homeassistant.components.squeezebox.config_flow.TIMEOUT", 0.1):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "no_server_found"}
+
+        # simulate manual input of host
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_HOST: HOST2}
+        )
+        assert result2["type"] == RESULT_TYPE_FORM
+        assert result2["step_id"] == "edit"
+        assert CONF_HOST in result2["data_schema"].schema
+        for key in result2["data_schema"].schema:
+            if key == CONF_HOST:
+                assert key.description == {"suggested_value": HOST2}
+
+
+async def test_user_form_duplicate(hass):
+    """Test duplicate discovered servers are skipped."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover", mock_discover,
+    ), patch("homeassistant.components.squeezebox.config_flow.TIMEOUT", 0.1), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.squeezebox.async_setup_entry", return_value=True,
+    ):
+        entry = MockConfigEntry(domain=DOMAIN, unique_id=UUID)
+        await hass.config_entries.async_add(entry)
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "no_server_found"}
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "edit"}
+    )
+
+    async def patch_async_query(self, *args):
+        self.http_status = HTTP_UNAUTHORIZED
+        return False
+
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: HOST,
+                CONF_PORT: PORT,
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_auth"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)